ppa/archive/management/commands/index.py

```diff
@@ -0,0 +1,154 @@
+'''
+**index** is a custom manage command to index PPA digitized work and page
+content into Solr.  It should be run _after_ content has been imported
+to the project database via the **hathi_import** manage command.
+
+Page content will be indexed from a local copy of dataset files under the
+configured **HATHI_DATA** path in pairtree format retrieved by rsync.
+(Note that pairtree data must include pairtree version file to be valid.)
+
+By default, indexes all content, digitized works and pages both.  You
+can optionally specify works or pages only, or index specific items
+by hathi id, but the id must exist in the database and the pairtree content
+for the items still must exist at the configured path.
+
+A progress bar will be displayed by default if there are more than 5
+items to process.  This can be suppressed via script options
+
+Example usage::
+
+    # index everything
+    python manage.py index
+    # index specific items
+    python manage.py index htid1 htid2 htid3
+    # index works only (skip pages)
+    python manage.py index -i works
+    # index pages only (skip works)
+    python manage.py index -i pages
+    # suppress progressbar
+    python manage.py index --no-progress
+
+'''
+
+
+import itertools
+
+from django.core.management.base import BaseCommand, CommandError
+from django.core.paginator import Paginator
+from django.db.models import Sum
+import progressbar
+# from urllib3.exceptions import HTTPError
+from SolrClient.exceptions import ConnectionError
+from requests.exceptions import RequestException
+
+from ppa.archive.models import DigitizedWork
+from ppa.archive.solr import get_solr_connection
+
+
+class Command(BaseCommand):
+    '''Reindex digitized items into Solr that have already been imported'''
+    help = __doc__
+
+    solr = None
+    solr_collection = None
+    bib_api = None
+    hathi_pairtree = {}
+    stats = None
+
+    options = {}
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+
+    #: solr params for index call; currently set to commit within 5 seconds
+    solr_index_opts = {"commitWithin": 5000}
```

Comment:
random thought: you could make this a CLI argument too? not sure how much it matters, but if differences in RAM across machines actually do affect it substantially, this would allow you to play around until you got a workable solution

Reply:
Nice idea.  Option should be in seconds probably?  I tried 5 and 3 and didn't see much difference between them for my dev setup.

Reply:
I like the idea of this being a command line argument, though the good thing is that the current number did work on the default (minimal) JVM heap. So we may just have a good default already.

Reply:
@rlskoeser that sounds right. and definitely agree @bwhicks 5sec can be the default